### PR TITLE
ci: E2E Framework [Scenarios] [5/6]

### PR DESCRIPTION
**Pull request by matmerr**
**Reason for Change**:

Has strong dependency on:

https://github.com/Azure/azure-container-networking/pull/2528
https://github.com/Azure/azure-container-networking/pull/2529

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

File: test/e2e/scenarios/hubble/steps/validate-drop-metric.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
+package steps
```
Comment (jpayne3506) [inline test/e2e/scenarios/hubble/steps/validate-drop-metric.go:1]: Are all steps going to be listed under the same package `steps`? If not should this be under the `hubble` package?

File: test/e2e/scenarios/hubble/steps/validate-drop-metric.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+	return metrics, nil
+}
+
+func parseReaderPrometheusMetrics(input io.Reader) (map[string]*promclient.MetricFamily, error) {
+	var parser expfmt.TextParser
+	return parser.TextToMetricFamilies(input) //nolint
+}
+
+func (v *ValidateHubbleDropMetric) Prevalidate() error {
```
Comment (jpayne3506) [inline test/e2e/scenarios/hubble/steps/validate-drop-metric.go:114]: nit: the only way to differentiate between `Prevalidate()`'s is based on how it is instantiated. Is there any other way we can define these functions as they coexist within the same package.

Comment (jpayne3506) [reply test/e2e/scenarios/hubble/steps/validate-drop-metric.go:114]: Also, what is the intent of the pre|post validate functions? Could this be a part of the [validate package](https://github.com/Azure/azure-container-networking/tree/master/test/validate)?

**Review by jpayne3506 — COMMENTED** [review]
Had a conversation offline about the E2E PR train global const(s) starting to run rampant. Once the train is complete there will be revisions to address.

File: test/e2e/scenarios/hubble/steps/validate-drop-metric.go (hunk 1 above)
Comment (matmerr) [reply test/e2e/scenarios/hubble/steps/validate-drop-metric.go]: Now that everything is here, scenarios and corresponding steps have been moved to respective subdirectories

File: test/e2e/scenarios/hubble/steps/validate-drop-metric.go (hunk 2 above)
Comment (matmerr) [reply test/e2e/scenarios/hubble/steps/validate-drop-metric.go]: The purpose of `Prevalidate()` is for quick tasks that can be done before execution to add guardrails before anything happens. After all steps have been added to the job, they're enumerated over and individually called `Prevalidate()`, as every step will now have fully populated properties to check

Additionally, can be used to ensure that files exist. For reference, see the Prevalidate in BYO cluster creation, ensuring cluster name length is <80chars, as well as ensuring deployment yamls exist in the embedded fs. 

File not found, and naming constraints are classic problems that show up 1hr into execution which can be easily avoided before anything is created

https://github.com/Azure/azure-container-networking/blob/faa2b736341e0ff0366ddfae5f9608cafca29a9e/test/e2e/framework/azure/create-cilium-cluster.go#L45-L69

**Review by vakalapa — APPROVED** [review]

**Review by vakalapa — APPROVED** [review]

**Review by vakalapa — APPROVED** [review]

**Pull request merged**